Libraries/src/Amazon.Lambda.DynamoDBEvents/DynamoDBEvent.cs

```diff
@@ -21,15 +18,315 @@ public class DynamoDBEvent
+            /// </li> 
+            /// <li> 
+            ///   <para>Records[].userIdentity.principalId</para>
+            ///   <para>"dynamodb.amazonaws.com"</para>
+            /// </li> 
+            /// </ul>
+            /// </summary>
+            public Identity UserIdentity { get; set; }
         }
 
+        /// <summary>
+        /// A description of a single data modification that was performed on an item in a DynamoDB table.
+        /// </summary>
+        public class StreamRecord
+        {
+            /// <summary>
+            /// The approximate date and time when the stream record was created, in <a href="http://www.epochconverter.com/">UNIX
+            /// epoch time</a> format and rounded down to the closest second.
+            /// </summary>
+            public DateTime ApproximateCreationDateTime { get; set; }
+
+            /// <summary>
+            /// The primary key attribute(s) for the DynamoDB item that was modified.
+            /// </summary>
+            public Dictionary<string, AttributeValue> Keys { get; set; }
+
+            /// <summary>
+            /// The item in the DynamoDB table as it appeared after it was modified.
+            /// </summary>
+            public Dictionary<string, AttributeValue> NewImage { get; set; }
+
+            /// <summary>
+            /// The item in the DynamoDB table as it appeared before it was modified.
+            /// </summary>
+            public Dictionary<string, AttributeValue> OldImage { get; set; }
+
+            /// <summary>
+            /// The sequence number of the stream record.
+            /// </summary>
+            public string SequenceNumber { get; set; }
+
+            /// <summary>
+            /// The size of the stream record, in bytes.
+            /// </summary>
+            public long SizeBytes { get; set; }
+
+            /// <summary>
+            /// <para>
+            /// The type of data from the modified DynamoDB item that was captured in this stream record:
+            /// </para>
+            /// 
+            /// <ul>
+            /// <li>
+            /// <para>
+            ///  <c>KEYS_ONLY</c> - only the key attributes of the modified item.
+            /// </para>
+            /// </li>
+            /// 
+            /// <li>
+            /// <para>
+            ///  <c>NEW_IMAGE</c> - the entire item, as it appeared after it was modified.
+            /// </para>
+            /// </li>
+            /// 
+            /// <li>
+            /// <para>
+            ///  <c>OLD_IMAGE</c> - the entire item, as it appeared before it was modified.
+            /// </para>
+            /// </li>
+            /// 
+            /// <li>
+            /// <para>
+            ///  <c>NEW_AND_OLD_IMAGES</c> - both the new and the old item images of the item.
+            /// </para>
+            /// </li>
+            /// </ul>
+            /// </summary>
+            public string StreamViewType { get; set; }

```

Comment:
This is an enum in the SDK, should we try to reproduce here? Though I suppose a plain .NET `enum` isn't as flexible as the generated constant class we use.

https://github.com/aws/aws-sdk-net/blob/459d2e5410af5a5b0e26ba6951e75bf0c420351a/sdk/src/Services/DynamoDBv2/Generated/Model/StreamRecord.cs#L180

Reply:
We have to use `string` to be forward compatible with new enum values and we don't have the `ConstantClass` to use here. Which the `ConstantClass` cause other fun JSON serialization issues.